File: 2279-maximum-bags-with-full-capacity-of-rocks/2279-maximum-bags-with-full-capacity-of-rocks.java
class Solution {
    public int maximumBags(int[] capacity, int[] rocks, int additionalRocks) {
        List<Integer> lst = new ArrayList();
        for (int i = 0; i < capacity.length; i++) {
            if (capacity[i] >= rocks[i]) {
                lst.add(capacity[i] - rocks[i]);
            }
        }
        Collections.sort(lst);
        int count = 0;
        int i = 0;
        while (i < lst.size() && additionalRocks > 0) {
            if (lst.get(i) <= additionalRocks) {
                additionalRocks -= lst.get(i);
                count++;
            }
            i++;
        }
        return count;
    }
}

File: 2279-maximum-bags-with-full-capacity-of-rocks/2279-maximum-bags-with-full-capacity-of-rocks.kt
class Solution {
    fun maximumBags(capacity: IntArray, rocks: IntArray, additionalRocks: Int): Int {
        val lst: MutableList<Int> = ArrayList()
        var moreRocks = additionalRocks
        for (i in capacity.indices) {
            if (capacity[i] >= rocks[i]) {
                lst.add(capacity[i] - rocks[i])
            }
        }
        lst.sort();
        var count = 0
        var i = 0
        while (i < lst.size && moreRocks > 0) {
            if (lst.get(i) <= moreRocks) {
                moreRocks -= lst.get(i)
                count++
            }
            i++
        }
        return count
    }
}